posts/2021/09/re-engineering-cicd-pipelines-for-scipy.md

```diff
@@ -0,0 +1,211 @@
+
+This blog post would assume a basic understanding of CI/CD and GitHub Actions. I will also assume a basic understanding of Python and the SciPy ecosystem.
+
+<p align="center">
+    <img
+     alt="The project will enable the developers to test out the all-new Meson build system on the GitHub Actions CI and lays down the further scope for developing an entire CI matrix for build, test and release for SciPy."
+     src="/images/2021/10/re-engineering-cicd-pipelines-for-scipy/re-engineering-ci-cd-scipy.png">
+    <i>Re-Engineering CI/CD pipelines for SciPy</i>
+</p>
+
+<!-- TEASER_END -->
+
+[SciPy](https://github.com/scipy/scipy) is quite an old and mature project, which is being used across the PyData community. The project currently uses a variety of continuous integration services and the Setuptools build system. As the project is growing and `distutils` will be deprecated as part of [PEP 632](https://www.python.org/dev/peps/pep-0632/), it was reasonable to migrate to a new build system. A [previous blog post](https://labs.quansight.org/blog/2021/07/moving-scipy-to-meson/) describes the migration process in detail and sets the future course for SciPy developers to look forward to faster builds.
+
+Scipy has been previously using [TravisCI](https://travis-ci.org/), [CircleCI](https://circleci.com/), and [AppVeyor](https://www.appveyor.com/) across its continuous integration pipelines. Recently, [GitHub Actions](https://github.com/features/actions), the native CI/CD service offered by GitHub, has been picked up by many projects. My project was centered around building and re-engineering the continuous integration pipelines for SciPy using GitHub Actions to be made compatible with the new build system.
+
+## Why GitHub Actions?
+
+GitHub Actions features a vast number of architectures, operating systems, runners, and more. With the aid of GitHub Actions, we can directly use the GitHub API from our workflows themselves. GitHub Actions also features a [vibrant marketplace](https://github.com/marketplace?type=actions) with applications centered around making the usage of GitHub Actions more straightforward for developers.
+
+We can also emulate local CI runs using a tool like [Act](https://github.com/nektos/act) to validate our GitHub Actions workflow. It allows us to run the entire workflow (including build and test) on our local machine, using a Docker image without initially testing it on the CI. Apart from this, whenever we fork a repository, the actions automatically get forked. It allows us to test on our fork and makes it relatively straightforward to use.
+
+<p align="center">
+    <img
+     alt="Meme depicting the state of CI service providers today. While many projects
```

Comment:
This also needs to be more descriptive. For example: "A meme showing two panels. In the top one, a man says "Life is good" while pointing at the camera. In the bottom panel, the same man smiles and says "But it can be better". Next to the meme, a series of logos for CI services are shown, including the logos for TravisCI, CircleCI and github actions" (I don't actually know all the logos!)